llvm/lib/CodeGen/GlobalISel/CombinerHelperArtifacts.cpp

```diff
@@ -33,10 +33,13 @@ bool CombinerHelper::matchMergeXAndUndef(const MachineInstr &MI,
   const GMerge *Merge = cast<GMerge>(&MI);
 
   Register Dst = Merge->getReg(0);
-  Register Undef = Merge->getSourceReg(1);
   LLT DstTy = MRI.getType(Dst);
   LLT SrcTy = MRI.getType(Merge->getSourceReg(0));
 
+  // Otherwise, we would miscompile.
+  if (Merge->getNumSources() > 2)
+    return false;
+
```

Comment:
It should still be assert, because this precondition should hold and if the assert breaks then we definitely need to know.